sat/cli/bootsys/bos.py

```diff
@@ -277,7 +277,10 @@ def has_completed(self):
             return False
 
         except APIError as err:
-            LOGGER.warning('Failed to query session status: %s', err)
+            if '404' in str(err):
```

Comment:
There is probably a better way to get the HTTP code from the response directly rather than looking at the string representation of the `APIError`. There is a keyword arg called `raise_not_ok` in `APIGatewayClient._make_req`. It defaults to `True`, but if it is set to `False`, it will not raise an `APIError` when it gets a non-ok (i.e. other than 2XX) HTTP response.

The line that raises the `APIError` here is line 249:

```
            self.session_status = self.bos_client.get_session_status(
                self.bos_session_thread.session_id
)
```

That method would have to be modified to accept and pass along the kwargs, so that we could pass `raise_not_ok=False`, and then we would be able to inspect the HTTP status code directly, which might be slightly more reliable. See here for an example of where we do this: https://github.com/Cray-HPE/sat/blob/7088e90a63d328324cfe41b863e237d91fc19faf/sat/cli/bootprep/input/session_template.py#L126-L132

Reply:
Thanks Ryan, for pointing to this approach. I think this lead me to cover more corner cases